spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/Fusion.scala

```diff
@@ -242,4 +231,46 @@ private[mkldnn] object Fusion {
       node.element.asInstanceOf[SpatialConvolution].negativeInput = false
     }
   }
+
+  def setJoinTableSame(node: Node[AbstractModule[Activity, Activity, Float]]): Unit = {
+    if (!fuse || !node.element.isInstanceOf[JoinTable]) return
+
+    val preConvs = node.prevNodes.flatMap(x => findAllNonIdentityPrevs(x))
+      .filter(_.element.isInstanceOf[SpatialConvolution])
+      .map(_.element.asInstanceOf[SpatialConvolution])
+
+    var needQuantize = preConvs.exists(_.needQuantize)
+    if (!needQuantize) return
+
+    val masks = preConvs.map(_.getOutputDimMask()).toSet
+    require(masks.size == 1, s"all the convolution should be the same mask")
+
+    val nextConvs = node.nextNodes.flatMap(findNext)
+      .filter(_.element.isInstanceOf[SpatialConvolution])
+
+    val scales = if (nextConvs.isEmpty) {
+      Array(preConvs.map(_.getOutputScales().flatten).transpose.map(_.max).toArray)
+    } else {
+      nextConvs.map(_.element.asInstanceOf[SpatialConvolution]).head.getInputScales()
+    }
+
+    preConvs.foreach { conv =>
+      conv.setOutputScales(scales)
+    }
+  }
+
+  private def findAllNonIdentityPrevs(node: Node[AbstractModule[Activity, Activity, Float]])
+  : Seq[Node[AbstractModule[Activity, Activity, Float]]] = {
+    // TODO currently, it will only skip the Identity, MaxPooling, AvgPooling, JoinTable
```

Comment:
TODO what ?

Reply:
We need to check whether the input of current node is not negative. So here we skip the previous nodes such as Identity, MaxPooling, AvgPooling And JoinTable. It can work but not an elegant way to implement.